codecov_auth/commands/owner/interactors/set_tokens_required.py

```diff
@@ -0,0 +1,43 @@
+from dataclasses import dataclass
+
+from codecov.commands.base import BaseInteractor
+from codecov.commands.exceptions import Unauthenticated, Unauthorized, ValidationError
+from codecov.db import sync_to_async
+from codecov_auth.helpers import current_user_part_of_org
+from codecov_auth.models import Owner
+
+
+@dataclass
+class SetTokensRequiredInput:
+    tokens_required: bool
+    org_username: str
+
+
+class SetTokensRequiredInteractor(BaseInteractor):
+    def validate(self, owner_obj):
+        if not self.current_user.is_authenticated:
+            raise Unauthenticated()
+        if not owner_obj:
+            raise ValidationError("Owner not found")
+        if not current_user_part_of_org(self.current_owner, owner_obj):
+            raise Unauthorized()
+        if not owner_obj.is_admin(self.current_owner):
+            raise Unauthorized("Admin authorization required")
+
+    @sync_to_async
+    def execute(self, input: dict):
+        typed_input = SetTokensRequiredInput(
+            tokens_required=input.get("tokens_required"),
+            org_username=input.get("org_username"),
+        )
+
+        owner_obj = Owner.objects.filter(
+            username=typed_input.org_username, service=self.service
+        ).first()
+
+        self.validate(owner_obj)
+
+        owner_obj.tokens_required = typed_input.tokens_required
```

Comment:
Thoughts on this being token_required? And even to be more explicit, `upload_token` required? 

We use the github/providers token for other things, so differentiating this as the "codecov_token" or "upload_token" would make more sense. I also don't know if this exclusively a coverage token or if it also applies to bundle_analysis/test_results, so I'd confirm what's the expected behavior, and if it is exclusively for coverage (and we don't intend to do it for BA/TR), then I'd call it `coverage_token` or `upload_coverage_token` or something like that

Reply:
This is just a draft PR not the final code, just put `tokens_required` as it's the suggested name by Trent for Nora, so  i will change the name according to what it's called in the DB to avoid any confusion

Reply:
Sounds good! 👌

Reply:
changed to upload_token_required!